Formula/atop.rb

```diff
@@ -0,0 +1,48 @@
+class Atop < Formula
+  desc "Advanced system and process monitor for Linux using process events"
+  homepage "https://www.atoptool.nl"
+  url "https://github.com/Atoptool/atop/archive/refs/tags/v2.6.0.tar.gz"
+  sha256 "134de16cd58fb651d952e8ac401b20fe7f724afc0b152f06486e5c6f7be9c31a"
+  license "GPL-2.0-or-later"
+  head "https://github.com/Atoptool/atop.git", branch: "master"
+
+  depends_on :linux
+  depends_on "ncurses"
+  depends_on "zlib"
+
+  def install
+    if build.head?
+      inreplace "version.h" do |s|
+        s.sub!(/"$/, "-#{Utils.git_short_head}\"")
+      end
+    end
+    # Some used paths are hard-coded in several places, we have to replace them:
+    inreplace Dir["Makefile", "*.*"] do |s|
+      s.gsub!(%r{/var}, var.to_s, false)
+      s.gsub!(%r{/etc/default}, etc.to_s, false)
+    end
+    # The Makefile tries to run systemctl, but brew install blocks access to systemd.
+    # We have to fake successful calls:
+    inreplace "Makefile", "systemctl", "true"
+    inreplace "Makefile", "chown root", "true"
```

Comment:
Why does it try and run systemctl?

Reply:
Because it tries to add a system activity recording daemon as a systemd service unit.
This background-recording for the atopsar command might be nice but it is not the core functionality of atop.
The Makefile tries to install and activate the that systemd service without asking, which is not that nice anyway.
The user can install the service himself if he needs it. It is delivered with the brew install command the user can activate it.

Reply:
That should be reported upstream and patched because in very few cases the packaging system is the same as the install system.

Reply:
Do you mean sysvinit vs. systemd?

I am a package Linux software package developer (distribution package developer) since 20 years, and as long as I remember, packages have a option where the package developer tells the package if it does systemd or sysvinit.

The age of sysvinit is long gone, most, especially the major (Debian/Ubuntu, Fedora, SUSE) Linux distributions have switched to systemd.

Bases don that I do not get what you want to say. We don't know if the user uses systemd, but it is >99% likely he does.
If not, he is using likely an embedded distribution and can just install the service his own way.

Reply:
I mean that it makes no sense to enable a service in the Makefile because the RPM for example will most likely be installed on another computer than the build computer where the user will still need to enable the service manually.

Reply:
Thanks for getting specific, you talk about RPM.

This is very simple: atop's source provides an example RPM spec file which every RPM package developer can use as basis. This is very helpful for RPM package developers like I was for 10 years.

The RPM and Debian packages of atop just simply do not use the Makefile to install the files,
the "make systemdinstall" rule and not even the "make genericinstall" is used by the example atop.spec.

This is actually common practice as often, the installation has to be adapted for the specific distribution.

Especially those non-autotools, purely Makefile-based projects often have Makefiles which can be used to build 
The Linux distributions have to support different versions of theire distro, e.g. on Ubuntu 21.04, /bin and /sbin are
only symlinks to /usr/bin and /usr/sbin, so such distros already differ from the Makefile. Sure lots of this can be done
automatically, but e.g. paths in the systemd services should better match the real paths, not the symlinks, which might
be removed at some point. Linux distro packaging is IMHO a Job that a OSS developer does not have to deal with,
he or her can't make everyone happy. Then ther are the Embeded Cross compilation build systems like OpenEmbdded,
which have yet again different needs. If the common user, it might be nice that the Makefile does everthing for the local system.

As said, making a distribution package is not something the normal OSS developer knows a lot about. I know it from 20 years.
As proof, just compare Fedora's atop.spec with the one from the atop source:

https://src.fedoraproject.org/rpms/atop/blob/77a41b3f4c2bdb440d312f7a3dcfce9e896284a1/f/atop.spec
https://github.com/Atoptool/atop/blob/master/rpmspec/atop.specsystemd
https://github.com/Atoptool/atop/blob/master/rpmspec/atop.specsysv

They look very very different. I do not expect an OSS developer to do the work of all the distros out there for them.

I am just volunteering here too, like you and right now I don't have many incentives to not install an atop service!

Just the command is enough, thanks. And I don't need it. I just thought I'd share the formula. And that's it.